lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -74,6 +105,64 @@ newTransactionLayer _proxy _protocolMagic = TransactionLayer
     , validateSelection = notImplemented "validateSelection"
     , allowUnbalancedTx = notImplemented "allowUnbalancedTx"
     }
+  where
+    _mkStdTx
+        :: (Address -> Maybe (k 'AddressK XPrv, Passphrase "encryption"))
+        -> SlotId
+        -> [(TxIn, TxOut)]
+        -> [TxOut]
+        -> Either ErrMkTx (Tx, SealedTx)
+    _mkStdTx keyFrom slot ownedIns outs = do
+        let ins' = map (toCardanoTxIn . fst) ownedIns
+        let outs' = map toCardanoTxOut outs
+        let fee = Coin $ sum (map (getCoin . coin . snd) ownedIns)
+                - sum (map (getCoin . coin) outs)
+        let certs = []
+
+        -- TODO: 1. Get the current epoch length
+        -- TODO: 2. The epoch length can change in the era change between byron and
+        -- shelley? It would probably be better to retrive a @SlotNo@ directly
+        -- without converting from @SlotId@.
+        let epochLength = EpochLength 1215
+
+        let Cardano.TxUnsignedShelley unsigned = Cardano.buildShelleyTransaction
```

Comment:
Would be nicer to use [`signTransaction`](https://github.com/input-output-hk/cardano-node/blob/c9ce637a260f415c8af21d75ddc6a1bb2efac23b/cardano-api/src/Cardano/Api.hs#L482-L496).

But I thought they way cardano-crypto carries an _encrypted_ XPrv and a password to the very end was intentional for… some securety reasons? To construct `SignKeyDSIGN` we'd need to provide the unencrypted key.

Reply:
>  was intentional for… some security reasons

Partly. Having key encrypted in-memory was an explicit request from the previous development team which was several time depicted as useless by cryptographers. What matters is storing them encrypted (cf the wallet-chain-libs rust implementation). We still use cardano-crypto for lack of alternative.